crates/matrix-sdk-crypto/src/verification/requests.rs

```diff
@@ -336,7 +336,21 @@ impl VerificationRequest {
 
         if let Some(future) = fut {
             let qr_verification = future.await?;
-            self.verification_cache.insert_qr(qr_verification.clone());
+
+            // We may have previously started our own QR verification (e.g. two devices
+            // displaying QR code at the same time), so we need to replace it with the newly
+            // scanned code.
+            if self
+                .verification_cache
+                .get_qr(qr_verification.other_user_id(), qr_verification.flow_id().as_str())
+                .is_some()
+            {
+                info!("Replacing existing QR verification");
```

Comment:
Not sure if those log lines should be `info()` they sound like `debug!()` to me, besides that a bit structure could be added to them, i.e. which user are we verifying, what's the flow id.

For example:

```suggestion
                info!(user_id = %self.other_user(), flow_id = self.flow_id().as_str(), "Replacing existing QR verification");

```

Note that this line is too long, but I didn't split it up fearing that the suggestion might break.